Services/Administration/OJS.Services.Administration.Business/Implementations/ZippedTestsParserService.cs

```diff
@@ -39,10 +42,122 @@ public async Task<TestsParseResult> Parse(Stream stream)
     }
 
     public int AddTestsToProblem(Problem problem, TestsParseResult tests)
-        => throw new System.NotImplementedException();
+    {
+        var tryOptimizeMysqlQuery = ShouldTryOptimizeMysqlQuery(problem);
+
+        var lastTrialTest = problem.Tests
+            .Where(x => x.IsTrialTest)
+            .OrderByDescending(x => x.OrderBy)
+            .FirstOrDefault();
+
+        var zeroTestsOrder = 1.0;
+
+        if (lastTrialTest != null)
+        {
+            zeroTestsOrder = lastTrialTest.OrderBy + 1;
+        }
+
+        var addedTestsCount = 0;
+
+        tests.ZeroInputs.ForEach((index, zeroInput) =>
+        {
+            problem.Tests.Add(new Test
+            {
+                IsTrialTest = true,
+                OrderBy = zeroTestsOrder,
+                Problem = problem,
+                InputDataAsString = GetInputData(zeroInput, tryOptimizeMysqlQuery),
+                OutputDataAsString = tests.ZeroOutputs[index],
+            });
```

Comment:
`problem.Tests.AddRange(tests.ZeroInputs.Select(...))`

Reply:
This does not apply here as i need the `index ` variable in my logic